### PR TITLE
add arrow docs, reset

### PR DESCRIPTION
**Pull request by joshrule**
Hmm... my git-fu appears to be a bit subpar. I was hoping I could perform some sort of rebase without closing the previous request. In any case, this one should be ready to merge.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -392,8 +392,8 @@ impl<N: Name> fmt::Display for TypeSchema<N> {
 
 /// Represents [monotypes][1] (fully instantiated, unquantified types).
 ///
-/// The primary ways of creating a `Type` are with the [`tp!`] macro or with
-/// [`TypeSchema::instantiate`].
+/// The primary ways of creating a `Type` are with the [`tp!`] macro,
```
Comment (lucasem) [inline src/lib.rs]: There are some missing words and punctuation around here

File: src/lib.rs [hunk 2]
```diff
@@ -430,6 +430,49 @@ pub enum Type<N: Name = &'static str> {
...
+    /// # fn main() {
+    /// use std::collections::VecDeque;
+    ///
+    /// let v = vec![tp!(int), tp!(int), tp!(bool)];
+    /// let t = Type::from(v);
+    /// assert_eq!(t.to_string(), "int → int → bool");
+    ///
+    /// let v = VecDeque::from(vec![tp!(int), tp!(int), tp!(bool)]);
+    /// let t = Type::from(v);
+    /// assert_eq!(t.to_string(), "int → int → bool");
```
Comment (lucasem) [inline src/lib.rs]: The VecDeque test is redundant for the docs, I think we should get rid of it.

**Comment by lucasem** [comment]
If you just rebase and force push to your branch, GitHub should be able to pick that up appropriately. No big deal, though

**Comment by joshrule** [comment]
Ah, I didn't force the push... I saw the error, deleted the remote branch, and then pushed a new version of it.

**Pull request merged**